spec/components/contents_list_with_body_spec.rb

```diff
@@ -0,0 +1,56 @@
+RSpec.describe "ContentsListWithBodyComponent", type: :view do
+  def component_name
+    "contents_list_with_body"
+  end
+
+  def contents_list
+    [
+      { href: "/one", text: "1. One" },
+      { href: "/two", text: "2. Two" },
+    ]
+  end
+
+  def block
+    "<p>Foo</p>".html_safe
+  end
+
+  it "renders nothing without a block" do
+    render_component(contents: contents_list)
+    expect(rendered).to be_empty
+  end
+
+  it "yields the block without contents data" do
+    render_component({}) { block }
+    expect(rendered).to include(block)
+  end
+
+  it "renders a sticky-element-container" do
+    render_component(contents: contents_list) { block }
+
+    expect(rendered).to have_css("#contents.app-c-contents-list-with-body")
+    expect(rendered).to have_css("#contents.app-c-contents-list-with-body")
+    expect(rendered).to have_css("#contents[data-module='sticky-element-container']")
+  end
+
+  it "does not apply the sticky-element-container data-module without contents data" do
+    render_component(contents: contents_list) { block }
+
+    expect(rendered).to have_css("#contents[data-module='sticky-element-container']")
```

Comment:
Is this test right? Test text says "does not apply the data module", but the expectation seems to thing it would still be there.

Reply:
Ah good spot thanks both! I copied the test over from government-frontend incorrectly - https://github.com/alphagov/government-frontend/blob/842a887dc0c83311a9ead2daa08bd430ad0ac347/test/components/contents_list_with_body_test.rb#L34-L37